### PR TITLE
Add REST Client Library Template

### LINKED ISSUE #80: Add template for REST client library
We have nice templates for a bunch of API servers, but none for API clients.

Consider leveraging the work in https://github.com/pyansys/openapi-common and the sketch at https://github.com/pyansys/template-openapi to make standard clients in a repeatable way.

### PR DESCRIPTION
**Pull request by da1910**
Closes #80

This PR adds a REST Client Library Template

Key features:
- Code generation is via CI/CD
- Generates _mostly_ PEP8 compliant python3 code
- Comes with built-in authentication support

Notes:
- Requires some repo secrets and authorizations for the template to work
- Likely will need some polishing

### CODE REVIEW

**Comment by jorgepiloto** [comment]
Hi @da1910, thank you for opening this. I fixed the description of the template so the "smoke tests" are passing now, check to commit [1994313](https://github.com/pyansys/ansys-templates/pull/85/commits/1994313c8cefc3d12b93462f8bf51d06d91c5238).

If you need help adding this to the CI so the baked template gets checked, let me now.

File: tests/tests_templates/test_python_templates.py [hunk 1]
```diff
@@ -190,8 +217,15 @@
 @pytest.mark.parametrize("build_system", ["flit", "poetry", "setuptools"])
 @pytest.mark.parametrize("template", TEMPLATES_VARIABLES_AND_STRUCTURE)
 def test_template_python(tmp_path, build_system, template):
+
+    # Get the list of supported build systems for the template
+    template_path = TEMPLATE_PATH_FINDER[template]
+    with open(template_path / "cookiecutter.json", 'r', encoding="utf-8") as fp:
+        config_json = json.load(fp)
+    supported_build_systems = config_json.get("build_system", ["setuptools"])
```
Comment (jorgepiloto) [inline tests/tests_templates/test_python_templates.py:225]: Beautiful! ❤️

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
This looks really good @da1910, thanks for all the effort. I agree with all changes you made.

It would be interesting to add this to the CI so this new template has a demo branch in the repo as it happens with the rest of the templates.

This logic is implemented in https://github.com/pyansys/ansys-templates/blob/main/.github/workflows/templates_python.yml

**Comment by da1910** [comment]
I'm not sure what the nicest way to exclude this template from the "test baked project step" is, or at least we can't install it with pip...

I suppose we could include a sample yaml file and then use maven to build it, do you think it's worth doing that?

File: src/ansys/templates/python/pyansys_openapi_client/{{cookiecutter.__project_name_slug}}/.github/workflows/generate_library.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Generate Library and Create PR
+
+on:
+  push:
+    tags:
+    - "*"
+    branches:
+    - main
+    paths:
+    - 'yaml/*.yaml'
```
Comment (da1910) [inline src/ansys/templates/python/pyansys_openapi_client/{{cookiecutter.__project_name_slug}}/.github/workflows/generate_library.yml:10]: use cookiecutter yaml name

File: .github/workflows/templates_python.yml [hunk 3]
```diff
@@ -37,6 +37,9 @@ jobs:
           - {name: "pyansys-advanced-flit", template: "pyansys_advanced", build-system: "flit", outdir: "pyansys_a0/pyproduct-library"}
           - {name: "pyansys-advanced-poetry", template: "pyansys_advanced", build-system: "poetry", outdir: "pyansys_a1/pyproduct-library"}
           - {name: "pyansys-advanced-setuptools", template: "pyansys_advanced", build-system: "setuptools", outdir: "pyansys_a2/pyproduct-library"}
+          
+          # PyAnsys OpenAPI Client template tests
+          - {name: "pyansys-openapi-client", template: "pyansys_openapi_client", build-system: "flit", outdir: "pyansys_o0/pyproduct-library-openapi"}
```
Comment (jorgepiloto) [inline .github/workflows/templates_python.yml:42]: I guess we can exclude this template from the CI so it can get merged, right @da1910?

Comment (da1910) [reply .github/workflows/templates_python.yml:42]: For the moment that's probably the best way forward. I would like to get an end-to-end test of the library, and a rendered example branch, but I haven't had much time recently.

Comment (da1910) [reply .github/workflows/templates_python.yml]: I've done so, I think this should be good to go for the moment

**Review by jorgepiloto — APPROVED** [review]
LGTM. Is there anything else you would like to add or modify, @da1910?

**Comment by da1910** [comment]
I think this should be good as a starting point!

**Comment by jorgepiloto** [comment]
Thanks for all the effort @da1910 and sorry for the delay merging this! 🚀

**Pull request merged**